# Assignment Week 11

[[_TOC_]]

## Tasks 
- Get a large, heterogeneous (regarding date, location, ...) set of nCoV(novel coronavirus) genomic sequences.
- Create a “classical” phylogenetic tree.
- Create a phylogenetic trees based on the method by Liao et al.[30]
- Compare the trees

## Dependencies
- python3 with:
    - jupyter notebooks
    - Biopython
    - matplotlib
    - pandas
    - numpy
    - seaborn
    - sklearn

## Data structure
- [MSA_and_sequences47.zip](https://git.imp.fu-berlin.de/verversl98/dsls-2020/-/blob/master/Week%2011/Paul/MSA_and_sequences47.zip) contains the "MSA47.fasta" multiple sequence alignment, the "sequences47.fasta" unaligned sequences and the "sequences47.csv" metadata from the NCBI Virus Database SARS-CoV-2 samples. The folder needs to be unzipped and the files need to be stored in the same directory as the notebook.

## How to run
- run the notebook [Sequence-curve_based_47_Sequences.ipynb](https://git.imp.fu-berlin.de/verversl98/dsls-2020/-/blob/master/Week%2011/Paul/Sequence-curve_based_47_Sequences.ipynb)
